algorithms/features/impl/return_value.py

```diff
@@ -5,126 +5,65 @@
-                    elif syscall.name() in self.not_interesting:
-                        pass
-                    else:
-                        print("not handled")
-                        print(syscall.name(), return_value_string)
-                except ValueError as e:
-                    if any(error in return_value_string for error in self.error_codes):
-                        # error code was returned so ValueError is expected
-                        # in calculateion -1 is returned
-                        pass
-                    else:
-                        print(e)
-                        print('Return Value: Could not cast return value to int')
-                        print(f' Return string: {return_value_string}')
-                        print(f' Syscall: {syscall.name()}')
+        return_value_string = syscall.param('res')
+        if return_value_string is not None:
+            try:
+                current_bytes = int(return_value_string)
+                if syscall.name() in self._max:
+                    if current_bytes >= self._max[syscall.name()]:
+                        self._max[syscall.name()] = current_bytes
+                else:
+                    self._max[syscall.name()] = current_bytes
+            except ValueError as e:
+                pass
 
     def _calculate(self, syscall: Syscall):
         """
         calculate return value type and normalize with max value of training phase
+        return -1 if 
+            * syscall never had return value in training
+            * return value was not an integer value
         """
         return_type = None
         normalized_bytes = 0
-        if syscall.name() in self.interesting:
-            return_value_string = syscall.param('res')
-            if return_value_string is not None:
-                try:
-                    current_bytes = int(return_value_string)
-                    if syscall.name() in self.write:
-                        return_type = 'write'
-                    elif syscall.name() in self.send_socket:
-                        return_type = 'send_socket'
-                    elif syscall.name() in self.read:
-                        return_type = 'read'
-                    elif syscall.name() in self.recv_socket:
-                        return_type = 'recv_socket'
-                    elif syscall.name() in self.get_dents:
-                        return_type = 'get_dents'
+        return_value_string = syscall.param('res')
+        if return_value_string is not None:
+            try:
+                current_bytes = int(return_value_string)
+            except ValueError as e:
+                return_type = 'not_int'
+                normalized_bytes = -1
+            try:
+                if return_type != 'not_int':
+                    if syscall.name() in self._max:
+                        if self._max[syscall.name()] != 0:
+                            normalized_bytes = current_bytes/self._max[syscall.name()]
```

Comment:
current_bytes might be referenced before assignment

Reply:
wenn das casten (int(return_value_string)) failed, dann wird ja return_type auf not_int gesetzt.
Ist das der Fall, wird current_bytes nicht mehr verwendent wegen Zeile 54.
Oder übersehe ich noch etwas?